meter/battery.go

```diff
@@ -0,0 +1,52 @@
+package meter
+
+import (
+	"github.com/evcc-io/evcc/api"
+)
+
+type battery struct {
+	MinSoc            float64
+	MaxDischargePower float64
+}
+
+// var _ api.BatteryController = (*battery)(nil)
+
+// Decorator returns a battery controller decorator
+func (m *battery) BatteryController(dischargePowerS, minSocS func(float64) error) (func() (api.BatteryMode, error), func(api.BatteryMode) error) {
+	get := func() (api.BatteryMode, error) {
+		return api.BatteryNormal, nil
+	}
+
+	set := func(mode api.BatteryMode) error {
+		switch mode {
+		case api.BatteryNormal:
+			// normal discharge
+			if err := dischargePowerS(m.MaxDischargePower); err != nil {
+				return err
+			}
+			// default minsoc
+			return minSocS(m.MinSoc)
+
+		case api.BatteryLocked:
+			// lock discharge
+			if err := dischargePowerS(0); err != nil {
+				return err
+			}
+			// default minsoc
+			return minSocS(m.MinSoc)
+
+		case api.BatteryCharge:
+			// normal discharge
+			if err := dischargePowerS(m.MaxDischargePower); err != nil {
```

Comment:
Das kann glaube ich weg, weil discharge und charge unterschiedlich gesetzt werden können. Und da hier der SoC auf 100% oder 95 gestzt wird spielt das keine Rolle. ggf. könnte man sogar discharge eher verbieten, da wir ja laden wollen.

Reply:
Das bleibt drin- wo sollte der Wert sonst zurück gestellt werden?

Reply:
Naja es ist ja eine statemachine. Wenn ich von state Charge in einen der anderen States geht wird es doch sofort auf den jeweils richtigen gesetzt. Aber ok lass ruhig so.